### ISSUE TITLE
[RFC] Should we maintain TypeScript type definitions in this repo?

### ISSUE THREAD

**Issue by nickuraltsev** [mzabriskie/axios#282]
axios TypeScript definitions are available on DefinitelyTyped: https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/axios

Is there any reason to maintain them in axios repository?

**Comment by kant2002** [comment]
I strongly think that - yes that should be in Axios repo. The reasons for that is evolving standards in TS ecosystem.
1. Deliver TS definitions inside NPM modules. That approach very convenient when you develop in TS initially. We could add 'typings:defeinition' property to `package.json`
2. TSD is now deprecated, and evolved as Typings project. That project has more robust system for delvering `.d.ts` files in the project then TSD. Better add `typings.json` to the repository to support and that scenario too.

**Comment by rluiten** [comment]
I am not sure this is appropriate place however it seems a good place to mention this so I dont loose my notes.

I noticed that typings installs definitions for 0.9.1, the one in this repository is 0.8.1, I do think its a good idea to have in the package, typings can then install it for people via the github site which reduces middle men etc.

In addition the following with the definition i had for 0.9.1 via typings utility.

1. interface `AxiosInstance` is missing `defaults` array. Which I needed when I was working with an extra `transformResponse` handler. Possibly I did things the wrong way, but I was grabbing the existing defaults so that I could append my transform to the end.
